src/tools/run-make-support/src/lib.rs

```diff
@@ -317,6 +317,26 @@ pub fn count_regex_matches_in_files_with_extension(re: &regex::Regex, ext: &str)
     count
 }
 
+/// Builds a static lib (`.lib` on Windows MSVC and `.a` for the rest) with the given name.
+#[track_caller]
+pub fn build_native_static_lib(lib_name: &str) -> PathBuf {
+    let obj_file = if is_msvc() { format!("{lib_name}") } else { format!("{lib_name}.o") };
+    let src = format!("{lib_name}.c");
+    let lib_path = static_lib_name(lib_name);
+    if is_msvc() {
+        cc().arg("-c").out_exe(&obj_file).input(src).run();
+    } else {
+        cc().arg("-v").arg("-c").out_exe(&obj_file).input(src).run();
+    };
+    let mut obj_file = PathBuf::from(format!("{lib_name}.o"));
+    if is_msvc() {
+        obj_file.set_extension("");
+        obj_file.set_extension("obj");
+    }
+    llvm_ar().obj_to_ar().output_input(&lib_path, &obj_file).run();
```

Comment:
Question: similar concern as https://github.com/rust-lang/rust/pull/126208#discussion_r1677102091, maybe let's block this PR on that?